### PR TITLE
Issue #787 Add tests for empty elements

### PR DESCRIPTION
**Pull request by lovemaths**

### CODE REVIEW

File: test/Microsoft.IdentityModel.Tests/ReferenceMetadata.cs [hunk 1]
```diff
@@ -725,6 +737,25 @@ public static string AADCommonMetadataFormated
             }
         }
 
+        public static string AADCommonMetadataWithEmptyElement
+        {
+            get
+            {
+                return
+                @"<EntityDescriptor xmlns=""urn:oasis:names:tc:SAML:2.0:metadata"" ID=""_6c4f3672-45c2-47a6-9515-afda95224009"" entityID=""https://sts.windows.net/268da1a1-9db4-48b9-b1fe-683250ba90cc/"">
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Tests/ReferenceMetadata.cs]: What I actually meant was that the %lt;EntityDescriptor> was empty.
I think this would fail if that was the case: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMetadataSerializer.cs#L135
Similar elements would also fail.
